eya_def_tools/eya_def_tools/reporting/reporting_engine.py

```diff
@@ -5,34 +5,86 @@
 
 """
 
-from pathlib import Path
-from typing import TypeAlias
-
 import pandas as pd
 
 from eya_def_tools.data_models import eya_def
+from eya_def_tools.data_models.wind_farm import WindFarmRelevance
+
+
+def generate_iec_table_a(eya_def_doc: eya_def.EyaDefDocument) -> pd.DataFrame:
+    iec_table_a = pd.DataFrame(
+        index=[
+            "Installed capacity [MW]",
+            "Export capacity [MW]",
+            "Number of turbines",
+            "Turbine model(s)",
+            "Turbine rated power [MW]",
+            "Turbine rotor diameter [m]",
+            "Turbine hub height [m]",
+        ],
+        columns=[scenario.label for scenario in eya_def_doc.scenarios],
+        dtype=str,
+    )
+
+    for scenario in eya_def_doc.scenarios:
+        internal_wind_farms = [
+            wind_farm
+            for wind_farm in eya_def_doc.wind_farms
+            if wind_farm.id in scenario.wind_farm_ids
+            and wind_farm.relevance == WindFarmRelevance.INTERNAL
+        ]
+        internal_turbines = [
+            turbine
+            for wind_farm in internal_wind_farms
+            for turbine in wind_farm.turbines
+        ]
+
+        installed_capacity = sum(
+            wind_farm.installed_capacity for wind_farm in internal_wind_farms
+        )
+
+        export_capacity = (
+            sum(
+                (
+                    wind_farm.export_capacity
+                    if wind_farm.export_capacity is not None
+                    else wind_farm.installed_capacity
+                )
+                for wind_farm in internal_wind_farms
+            )
+            if any(
+                wind_farm.export_capacity is not None
+                for wind_farm in internal_wind_farms
+            )
+            else ""
+        )
+
+        turbine_count = sum(
+            len(wind_farm.turbines) for wind_farm in internal_wind_farms
+        )
+
+        iec_table_a.at["Installed capacity [MW]", scenario.label] = (
+            f"{installed_capacity:.2f}"
+        )
 
-ReportingTableKey: TypeAlias = str  # TODO define enums
+        iec_table_a.at["Export capacity [MW]", scenario.label] = (
+            f"{export_capacity:.2f}"
+            if isinstance(export_capacity, float)
+            else str(export_capacity)
+        )
 
+        iec_table_a.at["Number of turbines", scenario.label] = str(turbine_count)
 
-class ReportingEngine:
-    """Engine for generating report tables from EYA DEF objects."""
+        iec_table_a.at["Turbine model(s)", scenario.label] = ""  # TODO
```

Comment:
The turbine model details depend on the IEC 61400-16 data model (Power Curve Schema) integration, which is another open PR. Those details can be added in the next iteration.